src/tv2-common/cues/EvaluateCueTelemetrics.ts

```diff
@@ -0,0 +1,82 @@
+import { IBlueprintPiece, IShowStyleUserContext, PieceLifespan, TSR } from '@tv2media/blueprints-integration'
+import { SharedOutputLayers } from '../../tv2-constants'
+import { AFVDSourceLayer } from '../../tv2_afvd_showstyle/layers'
+import { RobotCameraLayer } from '../../tv2_afvd_studio/layers'
+import { CalculateTime } from '../cueTiming'
+import { CueDefinitionTelemetrics } from '../inewsConversion'
+import { literal } from '../util'
+
+const TELEMETRICS_NAME_PREFIX: string = 'Robot'
+
+export function EvaluateCueTelemetrics(
+	_context: IShowStyleUserContext,
+	cueDefinition: CueDefinitionTelemetrics,
+	pieces: IBlueprintPiece[],
+	externalId: string
+): void {
+	const startTime: number = cueDefinition.start ? CalculateTime(cueDefinition.start) ?? 1 : 1
+
+	const existingPiece = findExistingPieceForExternalIdAndStartTime(pieces, externalId, startTime)
+	if (existingPiece) {
+		addPresetIdentifierToTimelineObject(existingPiece, cueDefinition.presetIdentifier)
+		addPresetIdentifierToPieceName(existingPiece, cueDefinition.presetIdentifier)
+		return
+	}
+
+	const newPiece = createTelemetricsPiece(externalId, cueDefinition, startTime)
+	pieces.push(newPiece)
+}
+
+function findExistingPieceForExternalIdAndStartTime(pieces: IBlueprintPiece[], externalId: string, startTime: number) {
+	return pieces.find(
+		piece =>
+			piece.externalId === externalId &&
```

Comment:
I wasn't sure if the externalId filter was needed or not since `EvaluateCues` just takes a Part, a PartDefinition and an array of Pieces. 
I think you are right we never evaluate cues for more than one Part at a time, but is there anything stopping us from writing code like this:
```
const part1 = createPart(1)
const part2 = createPart(2)
const pieces = []
const cues = [...]

EvaluateCues(part1, pieces, cues)
EvaluateCues(part2, pieces, cues)
```
In above example I think it would be necessary to evaluate on the externalId (above is hypothetical, but possible to do in our system). 

Will update with the Sourcelayer filter though

Reply:
Updated to filter on sourceLayer instead of externalId